### ISSUE TITLE
Bug with the new installer

### ISSUE THREAD

**Issue by slevyrosetti** [neuropoly/spinalcordtoolbox#775]
I just ran the new installer and got the following errors and warnings:
~~~
pass:~/spinalcordtoolbox $ ./install_sct 

Welcome to the SCT installer!

SCT version ......... 2.2_dev
Installation type ... git
Operating system .... osx

SCT will be installed here: [/Users/slevy_local/spinalcordtoolbox]
Do you agree? yes/no: yes

Skipping copy of source files (source and destination folders are the same)

Installing data...
curl -o data.tar.gz https://dl.dropboxusercontent.com/u/20592661/sct/MNI-Poly-AMU.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 62.4M  100 62.4M    0     0  8859k      0  0:00:07  0:00:07 --:--:--  9.8M
x data/
x data/atlas/
x data/colormap/
x data/gm_model/
x data/spinal_levels/
x data/template/
x data/template/bobo.lst
x data/template/info_label.txt
x data/template/landmarks_center.nii.gz
x data/template/mask_gaussian_templatespace_sigma20.nii.gz
x data/template/MNI-Poly-AMU_cord.nii.gz
x data/template/MNI-Poly-AMU_CSF.nii.gz
x data/template/MNI-Poly-AMU_GM.nii.gz
x data/template/MNI-Poly-AMU_level.nii.gz
x data/template/MNI-Poly-AMU_seg.nii.gz
x data/template/MNI-Poly-AMU_T2.nii.gz
x data/template/MNI-Poly-AMU_WM.nii.gz
x data/template/README.md
x data/template/test.nii.gz
x data/spinal_levels/info_label.txt
x data/spinal_levels/spinal_level_C3.nii.gz
x data/spinal_levels/spinal_level_C4.nii.gz
x data/spinal_levels/spinal_level_C5.nii.gz
x data/spinal_levels/spinal_level_C6.nii.gz
x data/spinal_levels/spinal_level_C7.nii.gz
x data/spinal_levels/spinal_level_C8.nii.gz
x data/spinal_levels/spinal_level_T1.nii.gz
x data/spinal_levels/spinal_level_T2.nii.gz
x data/spinal_levels/spinal_level_T3.nii.gz
x data/spinal_levels/spinal_level_T4.nii.gz
x data/spinal_levels/spinal_level_T5.nii.gz
x data/gm_model/dictionary_slices.pklz
x data/gm_model/info.txt
x data/gm_model/pca_data.pklz
x data/gm_model/tau_levels_0.txt
x data/gm_model/tau_levels_int.txt
x data/colormap/colourbar.nii.gz
x data/colormap/README.txt
x data/colormap/sct_renderhsv.lut
x data/colormap/sct_renderhsv.png
x data/colormap/sct_renderjet.lut
x data/colormap/sct_renderjet.png
x data/atlas/info_label.txt
x data/atlas/README.md
x data/atlas/WMtract__00.nii.gz
x data/atlas/WMtract__01.nii.gz
x data/atlas/WMtract__02.nii.gz
x data/atlas/WMtract__03.nii.gz
x data/atlas/WMtract__04.nii.gz
x data/atlas/WMtract__05.nii.gz
x data/atlas/WMtract__06.nii.gz
x data/atlas/WMtract__07.nii.gz
x data/atlas/WMtract__08.nii.gz
x data/atlas/WMtract__09.nii.gz
x data/atlas/WMtract__10.nii.gz
x data/atlas/WMtract__11.nii.gz
x data/atlas/WMtract__12.nii.gz
x data/atlas/WMtract__13.nii.gz
x data/atlas/WMtract__14.nii.gz
x data/atlas/WMtract__15.nii.gz
x data/atlas/WMtract__16.nii.gz
x data/atlas/WMtract__17.nii.gz
x data/atlas/WMtract__18.nii.gz
x data/atlas/WMtract__19.nii.gz
x data/atlas/WMtract__20.nii.gz
x data/atlas/WMtract__21.nii.gz
x data/atlas/WMtract__22.nii.gz
x data/atlas/WMtract__23.nii.gz
x data/atlas/WMtract__24.nii.gz
x data/atlas/WMtract__25.nii.gz
x data/atlas/WMtract__26.nii.gz
x data/atlas/WMtract__27.nii.gz
x data/atlas/WMtract__28.nii.gz
x data/atlas/WMtract__29.nii.gz
x data/atlas/WMtract__30.nii.gz
x data/atlas/WMtract__31.nii.gz

Create launchers for Python scripts
#!/bin/bash
$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )/sct_launcher $(basename $0).py $@
yes isct_check_detection
yes isct_convert_binary_to_trilinear
yes isct_minc2volume-viewer
yes isct_test_ants
yes isct_test_function
yes isct_warpmovie_generator
yes msct_gmseg_utils
yes msct_image
yes msct_multiatlas_seg
yes sct_apply_transfo
yes sct_average_data_within_mask
no sct_change_image_type
yes sct_check_atlas_integrity
yes sct_check_dependencies
yes sct_compute_ernst_angle
yes sct_compute_hausdorff_distance
yes sct_compute_mscc
yes sct_compute_mtr
yes sct_compute_snr
yes sct_concat_transfo
yes sct_convert
yes sct_create_mask
yes sct_crop_image
yes sct_denoise
no sct_denoising_onlm
yes sct_dice_coefficient
yes sct_dmri_compute_dti
yes sct_dmri_compute_snr
yes sct_dmri_concat_bvals
yes sct_dmri_concat_bvecs
yes sct_dmri_create_noisemask
no sct_dmri_eddy_correct
yes sct_dmri_get_bvalue
yes sct_dmri_moco
yes sct_dmri_separate_b0_and_dwi
yes sct_dmri_transpose_bvecs
yes sct_download_data
yes sct_extract_metric
yes sct_flatten_sagittal
yes sct_fmri_compute_tsnr
yes sct_fmri_moco
no sct_get_centerline
yes sct_image
no sct_invert_image
yes sct_label_utils
yes sct_label_vertebrae
yes sct_maths
yes sct_process_segmentation
yes sct_propseg
yes sct_register_multimodal
yes sct_register_to_template
yes sct_resample
yes sct_segment_graymatter
yes sct_smooth_spinalcord
yes sct_straighten_spinalcord
yes sct_testing
no sct_viewer
yes sct_warp_template
yes test_regist

Installing binaries...
curl -o sct_binaries.tar.gz https://dl.dropboxusercontent.com/u/20592661/sct/bin/osx/sct_binaries.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 58.1M  100 58.1M    0     0  8752k      0  0:00:06  0:00:06 --:--:-- 9189k
x ./._isct_ANTSLandmarksBSplineTransform
x isct_ANTSLandmarksBSplineTransform
x ./._isct_ComposeMultiTransform
x isct_ComposeMultiTransform
x ./._isct_antsApplyTransforms
x isct_antsApplyTransforms
x ./._isct_antsRegistration
x isct_antsRegistration
x ./._isct_antsSliceRegularizedRegistration
x isct_antsSliceRegularizedRegistration
x ./._isct_crop_image
x isct_crop_image
x ./._isct_dice_coefficient
x isct_dice_coefficient
x ./._isct_orientation3d
x isct_orientation3d
x ./._isct_propseg
x isct_propseg

Installing conda...
curl -o miniconda.sh https://dl.dropboxusercontent.com/u/20592661/sct/miniconda/osx/miniconda.sh
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 19.6M  100 19.6M    0     0  6508k      0  0:00:03  0:00:03 --:--:-- 6557k
bash miniconda.sh -p /Users/slevy_local/spinalcordtoolbox/python -b -f
PREFIX=/Users/slevy_local/spinalcordtoolbox/python
installing: _cache-0.0-py27_x0 ...
installing: python-2.7.11-0 ...
installing: conda-env-2.4.5-py27_0 ...
installing: openssl-1.0.2d-0 ...
installing: pycosat-0.6.1-py27_0 ...
installing: pyyaml-3.11-py27_1 ...
installing: readline-6.2-2 ...
installing: requests-2.9.0-py27_0 ...
installing: sqlite-3.8.4.1-1 ...
installing: tk-8.5.18-0 ...
installing: yaml-0.1.6-0 ...
installing: zlib-1.2.8-0 ...
installing: conda-3.19.0-py27_0 ...
installing: pycrypto-2.6.1-py27_0 ...
installing: pip-7.1.2-py27_0 ...
installing: wheel-0.26.0-py27_1 ...
installing: setuptools-18.8.1-py27_0 ...
Python 2.7.11 :: Continuum Analytics, Inc.
creating default environment...
installation finished.
WARNING:
    You currently have a PYTHONPATH environment variable set. This may cause
    unexpected behavior when running the Python interpreter in Miniconda2.
    For best results, please verify that your PYTHONPATH only points to
    directories of packages that are compatible with the Python interpreter
    in Miniconda2: /Users/slevy_local/spinalcordtoolbox/python
. /Users/slevy_local/spinalcordtoolbox/python/bin/activate /Users/slevy_local/spinalcordtoolbox/python
discarding /Users/slevy_local/spinalcordtoolbox/python/bin from PATH
prepending /Users/slevy_local/spinalcordtoolbox/python/bin to PATH

Installing Python dependencies...
Fetching package metadata: ....
Solving package specifications: ...........................................................
Package plan for installation in environment /Users/slevy_local/spinalcordtoolbox/python:

The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    jbig-2.1                   |                0          31 KB
    jpeg-8d                    |                1         137 KB
    libpng-1.6.17              |                0         184 KB
    mkl-11.3.1                 |                0        99.1 MB
    openssl-1.0.2g             |                0         3.0 MB
    sqlite-3.9.2               |                0         1.2 MB
    xz-5.0.5                   |                1         173 KB
    freetype-2.5.5             |                0         732 KB
    libtiff-4.0.6              |                1         736 KB
    decorator-4.0.9            |           py27_0          12 KB
    fastcache-1.0.2            |           py27_0          23 KB
    numpy-1.10.4               |           py27_0         3.0 MB
    pillow-3.1.1               |           py27_0         472 KB
    pyparsing-2.0.3            |           py27_0          63 KB
    pytz-2016.2                |           py27_0         176 KB
    qt-4.8.7                   |                1        39.0 MB
    requests-2.9.1             |           py27_0         604 KB
    setuptools-20.3            |           py27_0         453 KB
    sip-4.16.9                 |           py27_0         234 KB
    six-1.10.0                 |           py27_0          16 KB
    wheel-0.29.0               |           py27_0          81 KB
    conda-4.0.5                |           py27_0         185 KB
    cycler-0.10.0              |           py27_0          11 KB
    networkx-1.11              |           py27_0         1.1 MB
    pip-8.1.1                  |           py27_0         1.5 MB
    pyqt-4.11.4                |           py27_1         4.0 MB
    python-dateutil-2.5.1      |           py27_0         235 KB
    scipy-0.17.0               |      np110py27_0        12.9 MB
    sympy-0.7.6                |           py27_0         6.2 MB
    matplotlib-1.5.1           |      np110py27_0         5.9 MB
    pandas-0.16.2              |      np110py27_0         5.1 MB
    scikit-image-0.11.3        |      np110py27_0        16.2 MB
    ------------------------------------------------------------
                                           Total:       202.8 MB

The following NEW packages will be INSTALLED:

    cycler:          0.10.0-py27_0     
    decorator:       4.0.9-py27_0      
    fastcache:       1.0.2-py27_0      
    freetype:        2.5.5-0           
    jbig:            2.1-0             
    jpeg:            8d-1              
    libpng:          1.6.17-0          
    libtiff:         4.0.6-1           
    matplotlib:      1.5.1-np110py27_0 
    mkl:             11.3.1-0          
    networkx:        1.11-py27_0       
    numpy:           1.10.4-py27_0     
    pandas:          0.16.2-np110py27_0
    pillow:          3.1.1-py27_0      
    pyparsing:       2.0.3-py27_0      
    pyqt:            4.11.4-py27_1     
    python-dateutil: 2.5.1-py27_0      
    pytz:            2016.2-py27_0     
    qt:              4.8.7-1           
    scikit-image:    0.11.3-np110py27_0
    scipy:           0.17.0-np110py27_0
    sip:             4.16.9-py27_0     
    six:             1.10.0-py27_0     
    sympy:           0.7.6-py27_0      
    xz:              5.0.5-1           

The following packages will be UPDATED:

    conda:           3.19.0-py27_0 --> 4.0.5-py27_0      
    openssl:         1.0.2d-0      --> 1.0.2g-0          
    pip:             7.1.2-py27_0  --> 8.1.1-py27_0      
    requests:        2.9.0-py27_0  --> 2.9.1-py27_0      
    setuptools:      18.8.1-py27_0 --> 20.3-py27_0       
    sqlite:          3.8.4.1-1     --> 3.9.2-0           
    wheel:           0.26.0-py27_1 --> 0.29.0-py27_0     

Fetching packages ...
jbig-2.1-0.tar 100% |################################################################################################| Time: 0:00:00 332.08 kB/s
jpeg-8d-1.tar. 100% |################################################################################################| Time: 0:00:00   1.59 MB/s
libpng-1.6.17- 100% |################################################################################################| Time: 0:00:00   2.05 MB/s
mkl-11.3.1-0.t 100% |################################################################################################| Time: 0:00:17   6.07 MB/s
openssl-1.0.2g 100% |################################################################################################| Time: 0:00:00   8.31 MB/s
sqlite-3.9.2-0 100% |################################################################################################| Time: 0:00:00   5.01 MB/s
xz-5.0.5-1.tar 100% |################################################################################################| Time: 0:00:00   2.55 MB/s
freetype-2.5.5 100% |################################################################################################| Time: 0:00:00   4.13 MB/s
libtiff-4.0.6- 100% |################################################################################################| Time: 0:00:00   4.15 MB/s
decorator-4.0. 100% |################################################################################################| Time: 0:00:00  18.03 MB/s
fastcache-1.0. 100% |################################################################################################| Time: 0:00:00   1.04 MB/s
numpy-1.10.4-p 100% |################################################################################################| Time: 0:00:00   6.01 MB/s
pillow-3.1.1-p 100% |################################################################################################| Time: 0:00:00   4.88 MB/s
pyparsing-2.0. 100% |################################################################################################| Time: 0:00:00   1.44 MB/s
pytz-2016.2-py 100% |################################################################################################| Time: 0:00:00   1.98 MB/s
qt-4.8.7-1.tar 100% |################################################################################################| Time: 0:00:07   5.24 MB/s
requests-2.9.1 100% |################################################################################################| Time: 0:00:00   3.05 MB/s
setuptools-20. 100% |################################################################################################| Time: 0:00:00   3.30 MB/s
sip-4.16.9-py2 100% |################################################################################################| Time: 0:00:00   2.15 MB/s
six-1.10.0-py2 100% |################################################################################################| Time: 0:00:00 758.66 kB/s
wheel-0.29.0-p 100% |################################################################################################| Time: 0:00:00   1.25 MB/s
conda-4.0.5-py 100% |################################################################################################| Time: 0:00:00   2.07 MB/s
cycler-0.10.0- 100% |################################################################################################| Time: 0:00:00  18.53 MB/s
networkx-1.11- 100% |################################################################################################| Time: 0:00:00   4.57 MB/s
pip-8.1.1-py27 100% |################################################################################################| Time: 0:00:00   7.86 MB/s
pyqt-4.11.4-py 100% |################################################################################################| Time: 0:00:00   7.76 MB/s
python-dateuti 100% |################################################################################################| Time: 0:00:00   2.24 MB/s
scipy-0.17.0-n 100% |################################################################################################| Time: 0:00:01   8.42 MB/s
sympy-0.7.6-py 100% |################################################################################################| Time: 0:00:00  10.38 MB/s
matplotlib-1.5 100% |################################################################################################| Time: 0:00:00   7.21 MB/s
pandas-0.16.2- 100% |################################################################################################| Time: 0:00:00  10.08 MB/s
scikit-image-0 100% |################################################################################################| Time: 0:00:03   5.43 MB/s
Extracting packages ...
[      COMPLETE      ]|###################################################################################################################| 100%
Unlinking packages ...
[      COMPLETE      ]|###################################################################################################################| 100%
Linking packages ...
[      COMPLETE      ]|###################################################################################################################| 100%
Collecting nibabel==2.0.2 (from -r /Users/slevy_local/spinalcordtoolbox/install/requirements/requirementsPip.txt (line 1))
  Cache entry deserialization failed, entry ignored
  Downloading nibabel-2.0.2.tar.gz (3.4MB)
    100% |████████████████████████████████| 3.4MB 235kB/s 
Building wheels for collected packages: nibabel
  Running setup.py bdist_wheel for nibabel ... done
  Stored in directory: /Users/slevy_local/Library/Caches/pip/wheels/ed/ad/5a/9c45caa7c4899cebc4b6f4ad3cd146c728382fe860cba5d537
Successfully built nibabel
Installing collected packages: nibabel
Successfully installed nibabel-2.0.2
Cache entry deserialization failed, entry ignored

Installing external Python dependencies...
curl -o wheels.tar.gz https://dl.dropboxusercontent.com/u/20592661/sct/wheel/osx/wheels.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 4682k  100 4682k    0     0  1450k      0  0:00:03  0:00:03 --:--:-- 1674k
x ./._dipy-0.10.0.dev0-cp27-none-macosx_10_5_x86_64.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl
x dipy-0.10.0.dev0-cp27-none-macosx_10_5_x86_64.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl
x ./._ornlm-0.1-cp27-none-macosx_10_5_x86_64.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl
x ornlm-0.1-cp27-none-macosx_10_5_x86_64.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl
Processing ./dipy-0.10.0.dev0-cp27-none-macosx_10_5_x86_64.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl
Processing ./ornlm-0.1-cp27-none-macosx_10_5_x86_64.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl
Collecting nibabel>=1.2.0 (from dipy==0.10.0.dev0)
Installing collected packages: nibabel, dipy, ornlm
Successfully installed dipy-0.10.0.dev0 nibabel-2.0.2 ornlm-0.1
All requirement installed!

Do you want to add the sct_* script to your PATH environment? yes/no: yes


Run sct_check_dependencies...
SpinalCord Toolbox version 2.2_dev
SCT installed in /Users/slevy_local/spinalcordtoolbox
PYTHON /Users/slevy_local/spinalcordtoolbox/python/bin/python /Users/slevy_local/spinalcordtoolbox/python /Users/slevy_local/spinalcordtoolbox
running /Users/slevy_local/spinalcordtoolbox/scripts/sct_check_dependencies.py

Check which OS is running... 
.. osx (Darwin-12.2.1-x86_64-i386-64bit)
Check number of CPU cores...
.. Available: 8
.. Used by SCT: 8
Check RAM... 
Primary memory available: 8.00 gigabytes
Wired Memory:		1864 MB
Active Memory:		3166 MB
Inactive Memory:	3023 MB
Free Memory:		35 MB
Check which Python is running...
.. /Users/slevy_local/spinalcordtoolbox/python/bin/python
Check SCT path...
.. /Users/slevy_local/spinalcordtoolbox
Check SCT version... 
.. 2.2_dev
Check if scikit-image (0.11.3) is installed.........[OK]
Check if matplotlib (1.5.1) is installed............[OK]
Check if scipy (0.17.0) is installed................[OK]
Check if sympy (0.7.6) is installed.................[OK]
Check if numpy (1.10.4) is installed................[OK]
Check if pandas (0.16.2) is installed...............[OK]
Check if nibabel (2.0.2) is installed...............[OK]
Check if ornlm is installed.........................[OK]
Check if dipy is installed..........................[OK]
Check ANTs compatibility with OS ...................[FAIL]
SpinalCord Toolbox version 2.2_dev
SCT installed in /Users/slevy_local/spinalcordtoolbox
PYTHON /Users/slevy_local/spinalcordtoolbox/python/bin/python /Users/slevy_local/spinalcordtoolbox/python /Users/slevy_local/spinalcordtoolbox
running /Users/slevy_local/spinalcordtoolbox/scripts/isct_test_ants.py
  File "/Users/slevy_local/spinalcordtoolbox/scripts/isct_test_ants.py", line 2
    $( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )/sct_launcher $(basename $0).py $@
    ^
SyntaxError: invalid syntax
Check PropSeg compatibility with OS ................[OK]

INSTALLATION ERROR!
Please copy the historic of this Terminal (starting with the command install_sct) and paste it in the SCT Help forum under a new ticket:
https://sourceforge.net/p/spinalcordtoolbox/discussion/help/
~~~

**Comment by poquirion** [comment]
Il semble que ce soit une vielle version de BASH qui n'aime pas la notation `$(code)` we can use the tick notation

**Comment by jcohenadad** [comment]
@poquirion: the installer re-writes all files under /scripts and replace the content by the launcher commands (2 lines).
This is a major bug.

**Comment by jcohenadad** [comment]
problem came because of old alias files (launcher was written into scripts)
fixed in: ecd177f71d87fbd05386a9598f0e9952fd42d91b
